lib/taskana-core/src/main/java/pro/taskana/common/internal/JobMapper.java

```diff
@@ -36,11 +36,12 @@
   Integer insertJob(@Param("job") ScheduledJob job);
 
   @Select(
-      "<script> SELECT   JOB_ID, PRIORITY, CREATED, DUE, STATE, LOCKED_BY, LOCK_EXPIRES, TYPE, RETRY_COUNT, ARGUMENTS "
+      "<script> SELECT JOB_ID, PRIORITY, CREATED, DUE, STATE, LOCKED_BY, LOCK_EXPIRES, TYPE, RETRY_COUNT, ARGUMENTS "
           + "FROM SCHEDULED_JOB "
           + "WHERE STATE IN ( 'READY') AND (DUE is null OR DUE &lt; #{now}) AND (LOCK_EXPIRES is null OR LOCK_EXPIRES &lt; #{now}) AND RETRY_COUNT > 0 "
           + "ORDER BY PRIORITY DESC "
-          + "<if test=\"_databaseId == 'db2'\">with UR </if> "
+          + "FOR UPDATE "
+          + "<if test=\"_databaseId == 'db2'\">WITH RS USE AND KEEP UPDATE LOCKS</if> "
```

Comment:
technically this change is not necessary. The provided test works without it. Since I don't know what this does and since it was already implemented like this in the selectAndClaim case I copied this line aswell.